### ISSUE TITLE
test_memory_checker

### ISSUE THREAD

**Issue by lguohan** [Azure/sonic-buildimage#8430]
#### Steps to reproduce the issue:
rrun test_memory_check on dual tor platform

#### Describe the results you received:
```
2021-08-10T12:35:17.8742225Z ============================= test session starts ==============================
2021-08-10T12:35:17.8745670Z platform linux2 -- Python 2.7.17, pytest-4.6.5, py-1.10.0, pluggy-0.13.1
2021-08-10T12:35:17.8794187Z ansible: 2.8.12
2021-08-10T12:35:17.8796625Z rootdir: /azp/agent/_work/8/s/tests, inifile: pytest.ini
2021-08-10T12:35:17.8800241Z plugins: metadata-1.11.0, forked-1.3.0, xdist-1.28.0, repeat-0.9.1, html-1.22.1, ansible-2.2.2
2021-08-10T12:35:17.9490934Z collected 1 item
2021-08-10T12:35:17.9491856Z 
2021-08-10T12:38:32.4567373Z memory_checker/test_memory_checker.py::test_memory_checker[str2-7050cx3-acs-07] 
2021-08-10T12:38:32.4576732Z -------------------------------- live log call ---------------------------------
2021-08-10T12:38:32.4580078Z 12:38:32 __init__.pytest_runtest_call             L0039 ERROR  | Traceback (most recent call last):
2021-08-10T12:38:32.4583526Z   File "/usr/local/lib/python2.7/dist-packages/_pytest/python.py", line 1464, in runtest
2021-08-10T12:38:32.4586379Z     self.ihook.pytest_pyfunc_call(pyfuncitem=self)
2021-08-10T12:38:32.4589505Z   File "/usr/local/lib/python2.7/dist-packages/pluggy/hooks.py", line 286, in __call__
2021-08-10T12:38:32.4592152Z     return self._hookexec(self, self.get_hookimpls(), kwargs)
2021-08-10T12:38:32.4595103Z   File "/usr/local/lib/python2.7/dist-packages/pluggy/manager.py", line 93, in _hookexec
2021-08-10T12:38:32.4597870Z     return self._inner_hookexec(hook, methods, kwargs)
2021-08-10T12:38:32.4600816Z   File "/usr/local/lib/python2.7/dist-packages/pluggy/manager.py", line 87, in <lambda>
2021-08-10T12:38:32.4603486Z     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
2021-08-10T12:38:32.4606463Z   File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 208, in _multicall
2021-08-10T12:38:32.4609107Z     return outcome.get_result()
2021-08-10T12:38:32.4611713Z   File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 81, in get_result
2021-08-10T12:38:32.4614378Z     _reraise(*ex)  # noqa
2021-08-10T12:38:32.4617028Z   File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 187, in _multicall
2021-08-10T12:38:32.4619556Z     res = hook_impl.function(*args)
2021-08-10T12:38:32.4622257Z   File "/usr/local/lib/python2.7/dist-packages/_pytest/python.py", line 174, in pytest_pyfunc_call
2021-08-10T12:38:32.4624783Z     testfunction(**testargs)
2021-08-10T12:38:32.4627262Z   File "/azp/agent/_work/8/s/tests/memory_checker/test_memory_checker.py", line 232, in test_memory_checker
2021-08-10T12:38:32.4630076Z     consume_memory_and_restart_container(duthost, container_name, vm_workers, loganalyzer, marker)
2021-08-10T12:38:32.4632903Z   File "/azp/agent/_work/8/s/tests/memory_checker/test_memory_checker.py", line 148, in consume_memory_and_restart_container
2021-08-10T12:38:32.4635457Z     loganalyzer.analyze(marker)
2021-08-10T12:38:32.4638335Z   File "/azp/agent/_work/8/s/tests/common/plugins/loganalyzer/loganalyzer.py", line 282, in analyze
2021-08-10T12:38:32.4640896Z     self._verify_log(analyzer_summary)
2021-08-10T12:38:32.4643519Z   File "/azp/agent/_work/8/s/tests/common/plugins/loganalyzer/loganalyzer.py", line 88, in _verify_log
2021-08-10T12:38:32.4646103Z     raise LogAnalyzerError(result)
2021-08-10T12:38:32.4650812Z LogAnalyzerError: {'match_messages': {'/tmp/syslog.str2-7050cx3-acs-07.2021-08-10-12:38:25': []}, 'total': {'expected_match': 0, 'expected_missing_match': 3, 'match': 0}, 'match_files': {'/tmp/syslog.str2-7050cx3-acs-07.2021-08-10-12:38:25': {'expected_match': 0, 'match': 0}}, 'expect_messages': {'/tmp/syslog.str2-7050cx3-acs-07.2021-08-10-12:38:25': []}, 'unused_expected_regexp': [".*restart_service.*Restarting service 'telemetry'.*", '.*Stopping Telemetry container.*', '.*Stopped Telemetry container.*']}
2021-08-10T12:38:32.4653497Z 
2021-08-10T12:38:36.3274008Z 
2021-08-10T12:38:36.3275371Z 
2021-08-10T12:38:36.3278142Z =================================== FAILURES ===================================
2021-08-10T12:38:36.3281551Z ___________________ test_memory_checker[str2-7050cx3-acs-07] ___________________
2021-08-10T12:38:36.3282628Z 
2021-08-10T12:38:36.3284952Z duthosts = <tests.common.devices.duthosts.DutHosts object at 0x7f3d565d86d0>
2021-08-10T12:38:36.3288935Z creds = {'ASIC0': {'configuration': {'ASIC4': {'bgp': {'asn': 65100, 'peers': {65100: [...]}}, 'interfaces': {'Eth0-ASIC4': {'...'asic_intfs': ['Eth24-ASIC3', 'Eth25-ASIC3', 'Eth26-ASIC3', 'Eth27-ASIC3', 'Eth28-ASIC3', 'Eth29-ASIC3', ...]}}}}, ...}
2021-08-10T12:38:36.3292219Z enum_rand_one_per_hwsku_frontend_hostname = 'str2-7050cx3-acs-07'
2021-08-10T12:38:36.3293214Z 
2021-08-10T12:38:36.3295595Z     def test_memory_checker(duthosts, creds, enum_rand_one_per_hwsku_frontend_hostname):
2021-08-10T12:38:36.3298327Z         """Checks whether the telemetry container can be restarted or not if the memory
2021-08-10T12:38:36.3300795Z         usage of it is beyond the threshold. The `stress` utility is leveraged as
2021-08-10T12:38:36.3303055Z         the memory stressing tool.
2021-08-10T12:38:36.3305103Z     
2021-08-10T12:38:36.3307083Z         Args:
2021-08-10T12:38:36.3309123Z             duthosts: The fixture returns list of DuTs.
2021-08-10T12:38:36.3311604Z             enum_rand_one_per_hwsku_frontend_hostname: The fixture randomly pick up
2021-08-10T12:38:36.3313913Z               a frontend DuT from testbed.
2021-08-10T12:38:36.3315964Z     
2021-08-10T12:38:36.3318509Z         Returns:
2021-08-10T12:38:36.3320823Z             None.
2021-08-10T12:38:36.3322975Z         """
2021-08-10T12:38:36.3325220Z         duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
2021-08-10T12:38:36.3328275Z         # TODO: Currently we only test 'telemetry' container which has the memory threshold 400MB
2021-08-10T12:38:36.3331112Z         # and number of vm_workers is hard coded. We will extend this testing on all containers after
2021-08-10T12:38:36.3334141Z         # the feature 'memory_checker' is fully implemented.
2021-08-10T12:38:36.3336592Z         container_name = "telemetry"
2021-08-10T12:38:36.3345847Z         vm_workers = 4
2021-08-10T12:38:36.3347983Z     
2021-08-10T12:38:36.3350571Z         pytest_require(("20191130" in duthost.os_version and parse_version(duthost.os_version) > parse_version("20191130.72"))
2021-08-10T12:38:36.3353452Z                        or parse_version(duthost.kernel_version) > parse_version("4.9.0"),
2021-08-10T12:38:36.3356147Z                        "Test is not supported for 20191130.72 and older image versions!")
2021-08-10T12:38:36.3358874Z     
2021-08-10T12:38:36.3361013Z     
2021-08-10T12:38:36.3363244Z         expected_alerting_messages = []
2021-08-10T12:38:36.3365868Z         loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix="container_restart_due_to_memory")
2021-08-10T12:38:36.3368475Z         loganalyzer.expect_regex = []
2021-08-10T12:38:36.3371636Z         expected_alerting_messages.append(".*restart_service.*Restarting service 'telemetry'.*")
2021-08-10T12:38:36.3374376Z         expected_alerting_messages.append(".*Stopping Telemetry container.*")
2021-08-10T12:38:36.3377007Z         expected_alerting_messages.append(".*Stopped Telemetry container.*")
2021-08-10T12:38:36.3379514Z     
2021-08-10T12:38:36.3381839Z         loganalyzer.expect_regex.extend(expected_alerting_messages)
2021-08-10T12:38:36.3384602Z         marker = loganalyzer.init()
2021-08-10T12:38:36.3387001Z     
2021-08-10T12:38:36.3389381Z         install_stress_utility(duthost, creds, container_name)
2021-08-10T12:38:36.3392208Z >       consume_memory_and_restart_container(duthost, container_name, vm_workers, loganalyzer, marker)
2021-08-10T12:38:36.3393340Z 
2021-08-10T12:38:36.3395759Z container_name = 'telemetry'
2021-08-10T12:38:36.3400044Z creds      = {'ASIC0': {'configuration': {'ASIC4': {'bgp': {'asn': 65100, 'peers': {65100: [...]}}, 'interfaces': {'Eth0-ASIC4': {'...'asic_intfs': ['Eth24-ASIC3', 'Eth25-ASIC3', 'Eth26-ASIC3', 'Eth27-ASIC3', 'Eth28-ASIC3', 'Eth29-ASIC3', ...]}}}}, ...}
2021-08-10T12:38:36.3403376Z duthost    = <MultiAsicSonicHost> str2-7050cx3-acs-07
2021-08-10T12:38:36.3405994Z duthosts   = <tests.common.devices.duthosts.DutHosts object at 0x7f3d565d86d0>
2021-08-10T12:38:36.3408858Z enum_rand_one_per_hwsku_frontend_hostname = 'str2-7050cx3-acs-07'
2021-08-10T12:38:36.3412133Z expected_alerting_messages = [".*restart_service.*Restarting service 'telemetry'.*", '.*Stopping Telemetry container.*', '.*Stopped Telemetry container.*']
2021-08-10T12:38:36.3415080Z loganalyzer = <tests.common.plugins.loganalyzer.loganalyzer.LogAnalyzer instance at 0x7f3d55b33410>
2021-08-10T12:38:36.3417896Z marker     = 'container_restart_due_to_memory.2021-08-10-12:36:34'
2021-08-10T12:38:36.3420343Z vm_workers = 4
2021-08-10T12:38:36.3421094Z 
2021-08-10T12:38:36.3423290Z memory_checker/test_memory_checker.py:232: 
2021-08-10T12:38:36.3425816Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-08-10T12:38:36.3428517Z memory_checker/test_memory_checker.py:148: in consume_memory_and_restart_container
2021-08-10T12:38:36.3431017Z     loganalyzer.analyze(marker)
2021-08-10T12:38:36.3433413Z common/plugins/loganalyzer/loganalyzer.py:282: in analyze
2021-08-10T12:38:36.3435898Z     self._verify_log(analyzer_summary)
2021-08-10T12:38:36.3438559Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-08-10T12:38:36.3439642Z 
2021-08-10T12:38:36.3442178Z self = <tests.common.plugins.loganalyzer.loganalyzer.LogAnalyzer instance at 0x7f3d55b33410>
2021-08-10T12:38:36.3446106Z result = {'expect_messages': {'/tmp/syslog.str2-7050cx3-acs-07.2021-08-10-12:38:25': []}, 'match_files': {'/tmp/syslog.str2-705...7050cx3-acs-07.2021-08-10-12:38:25': []}, 'total': {'expected_match': 0, 'expected_missing_match': 3, 'match': 0}, ...}
2021-08-10T12:38:36.3447996Z 
2021-08-10T12:38:36.3450207Z     def _verify_log(self, result):
2021-08-10T12:38:36.3452338Z         """
2021-08-10T12:38:36.3454704Z         Verify that total match and expected missing match equals to zero or raise exception otherwise.
2021-08-10T12:38:36.3457836Z         Verify that expected_match is not equal to zero when there is configured expected regexp in self.expect_regex list
2021-08-10T12:38:36.3460375Z         """
2021-08-10T12:38:36.3462481Z         if not result:
2021-08-10T12:38:36.3465005Z             raise LogAnalyzerError("Log analyzer failed - no result.")
2021-08-10T12:38:36.3467728Z         if result["total"]["match"] != 0 or result["total"]["expected_missing_match"] != 0:
2021-08-10T12:38:36.3470371Z >           raise LogAnalyzerError(result)
2021-08-10T12:38:36.3474678Z E           LogAnalyzerError: {'match_messages': {'/tmp/syslog.str2-7050cx3-acs-07.2021-08-10-12:38:25': []}, 'total': {'expected_match': 0, 'expected_missing_match': 3, 'match': 0}, 'match_files': {'/tmp/syslog.str2-7050cx3-acs-07.2021-08-10-12:38:25': {'expected_match': 0, 'match': 0}}, 'expect_messages': {'/tmp/syslog.str2-7050cx3-acs-07.2021-08-10-12:38:25': []}, 'unused_expected_regexp': [".*restart_service.*Restarting service 'telemetry'.*", '.*Stopping Telemetry container.*', '.*Stopped Telemetry container.*']}
2021-08-10T12:38:36.3477655Z 
2021-08-10T12:38:36.3481142Z result     = {'expect_messages': {'/tmp/syslog.str2-7050cx3-acs-07.2021-08-10-12:38:25': []}, 'match_files': {'/tmp/syslog.str2-705...7050cx3-acs-07.2021-08-10-12:38:25': []}, 'total': {'expected_match': 0, 'expected_missing_match': 3, 'match': 0}, ...}
2021-08-10T12:38:36.3484600Z self       = <tests.common.plugins.loganalyzer.loganalyzer.LogAnalyzer instance at 0x7f3d55b33410>
2021-08-10T12:38:36.3485705Z 
2021-08-10T12:38:36.3487891Z common/plugins/loganalyzer/loganalyzer.py:88: LogAnalyzerError
2021-08-10T12:38:36.3490928Z - generated xml file: /azp/agent/_work/8/s/tests/logs/memory_checker/test_memory_checker.xml -
2021-08-10T12:38:36.3493604Z =========================== short test summary info ============================
2021-08-10T12:38:36.3496321Z FAILED memory_checker/test_memory_checker.py::test_memory_checker[str2-7050cx3-acs-07]
```

#### Output of `show version`:

```
(paste your output here)
```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

**Comment by yozhao101** [comment]
[Update]: I reproduced this issue on a Dual ToR `str2-7050cx3-acs-8`. The `test_memory_checker` will install a utility `stress` in the `streaming telemetry` container. But the `stress` utility sometimes can not be installed correctly. I am working on to figure out the root reason why the installation failed.

**Comment by yozhao101** [comment]
[Update]: This pytest script failed only on Dual ToR and passed all single ToR testing.

**Comment by yozhao101** [comment]
@lguohan @yxieca 

 [Update **Root Cause**]: 
The logic of this pytest script `test_memory_checker.py` is as following:
1. (Fixture function) Reduces the monitoring interval of Monit from 60 seconds to 10 seconds.
2. Installs the `stress` utility in `streaming telemetry` container.
3. Runs `stress` utility to consume more than 400MB memory.
4. Checks whether the `streaming telemetry` container will be restarted or not.

One sentence to summarize the root cause is the step 1 was executed on DuT `str2-7050cx3-08` and steps 2, 3, 4 were executed on another DuT `str2-7050cx3-acs-09`. I think this behavior was not expected.

In detail, I ran this pytest script against the testbed `vms17-dual-t0-7050-2`. The Dual ToR `str2-7050cx3-acs-8` was selected to run this test. Monit control file `monitrc` on this device `str2-7050cx3-acs-8` was changed such that Monit will check each monitoring item every 10 seconds. But the interesting thing is the `stress` utility was installed on another DuT `str2-7050cx3-acs-9` and then
`stress` utility was called to consume memory. Since `stress` utility was not installed and ran on DuT `str2-7050cx3-acs-8`, logAnalyzer was unable to find the corresponding syslog messages and then failed this pytest script.

I also checked the syslog of nightly test at the testbed `vms17-dual-t0-7050-1` on 8/16/2021 and found the same issue. The link of syslog is here: https://dev.azure.com/mssonic/internal/_build/results?buildId=29381&view=artifacts&pathAsName=false&type=publishedArtifacts.
